pkg/internal/export/otel/traces.go

```diff
@@ -370,7 +388,7 @@ func GenerateTraces(span *request.Span, userAttrs map[attr.Name]struct{}) ptrace
 	traces := ptrace.NewTraces()
 	rs := traces.ResourceSpans().AppendEmpty()
 	ss := rs.ScopeSpans().AppendEmpty()
-	resourceAttrs := attrsToMap(getAppResourceAttrs(&span.ServiceID))
+	resourceAttrs := attrsToMap(traceAppResourceAttrs(&span.ServiceID))
```

Comment:
I believe this is safe, since the attrsToMap makes a new map object out of the array.